src/goto-symex/symex_target_equation.h

```diff
@@ -291,14 +292,12 @@ class symex_target_equationt:public symex_targett
     SSA_stepst::iterator it=SSA_steps.begin();
     for(; s!=0; s--)
     {
-      assert(it!=SSA_steps.end());
+      PRECONDITION(it != SSA_steps.end());
       it++;
     }
     return it;
   }
 
-  void output(std::ostream &out) const;
-
```

Comment:
Because I removed the `ns` member, which is needed for that function. This was all part of my effort to clarify what classes own what objects, but I'm afraid I don't recall exactly why `ns` was a member of this class (perhaps @tautschnig could elaborate? I think we discussed this and I forgot...)

When I wanted to create the `branch_pointt` class (consisting of an equation and a state), I tried to remove all members that would point to objects that might become stale, since we need to resume the equation and state later and stale/out-of-scope references led to segfaults. I _think_ the `ns` member was one of the victims of this purge, and so was the `dirtyt` member (that's why I changed it from a pointer to a member)...

Reply:
I think we weren't able to figure out why `ns` should possibly be a member here. Indeed @martin-cs is right that `output` needs review. We should have a method that is explicitly passed a `namespacet` as an argument. That implies that the `operator<<` declarations should go away as well (their definitions have been removed!). I'd suggest to re-instantiate the `output` methods exactly as it was defined before, except that it should take an additional `namespact` parameter.

Reply:
OK, will push that in a few minutes